Packs/Cryptosim/Integrations/Cryptosim/Cryptosim.py

```diff
@@ -0,0 +1,226 @@
+        parameters = {
+            'startDate': args.get('startDate', start_time.isoformat()),
+            'endDate': args.get('endDate', end_time.isoformat()),
+            'showSolved': args.get('showSolved', False),
+            'crrPluginId': args.get('crrPluginId', -1),
+            'containStr': args.get('containStr', None),
+            'risk': args.get('risk', -1),
+            'srcIPPort': args.get('srcIPPort', None),
+            'destIPPort': args.get('destIPPort', None),
+            'srcPort': args.get('srcPort', None),
+            'destPort': args.get('destPort', None),
+            'riskOperatorID': args.get('riskOperatorID', "equal"),
+            "isJsonLog": True
+        }
+
+        return self._http_request("POST", url_suffix="correlationalertswithlogs",
+                                    data=json.dumps(parameters))
+
+    def correlations(self):
+        return self._http_request("GET", data={}, url_suffix="correlations")
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def correlations_command(client: Client):
+    result = client.correlations()
+
+    return CommandResults(
+        outputs_prefix='Correlations',
+        outputs_key_field='',
+        outputs=result,
+    )
+
+
+def correlation_alerts_command(client: Client):
+
+    # Call the Client function and get the raw response
+    result = client.correlation_alerts()
+
+    return CommandResults(
+        outputs_prefix='CorrelationAlerts',
+        outputs_key_field='',
+        outputs=result,
+    )
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+    :type client: ``Client``
+    :param Client: client to use
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        if client.correlations().get('StatusCode') == 200:
+            message = 'ok'
+        else:
+            raise Exception(f"""StatusCode: 
+                            {client.correlations().get('StatusCode')}, 
+                            Error: {client.correlations().get('ErrorMessage')}
+                            """)
+    except DemistoException as e:
+        if '401' in str(e):
+            message = 'Authorization Error: make sure API User and Password is correctly set'
+        else:
+            raise e
+    return message
+
+
+''' INCIDENT '''
+
+
+def fetch_incidents(client: Client):
+
+    max_results = arg_to_number(arg=demisto.params().get('max_fetch'), arg_name='max_fetch', required=False)
+
+
+    first_fetch_time = arg_to_datetime(demisto.params().get('first_fetch')).isoformat()
+
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get('last_fetch', first_fetch_time)
+
+    incidentsList=[]
+    alert_response = client.correlation_alerts(end_time=last_fetch)
```

Comment:
But the `last_fetch` is your new `start_time` not `end_time`

Reply:
Yes, 
In this case:
start_time is depends end_time as it comes from definition.
end_time is current date or given last_fetch time. Then, start_time is determined end_time - FetchIntervalTime.
Example:
fetch_time = "2022-03-16T16:55:00"
intervalFetch = "1 hour"
if fetch_incident trigger the correlation_alerts:
end_time will be fetch_time
and start time will be  "2022-03-16T15:55:00". So start time depends the last_fetch time

Reply:
> We want that when you are fetching, you will pass the start_time by the last incident that you got the last fetch

I changed end time is always datetime.now. but start_time is depend last_fetch and fetch interval.
```
def correlation_alerts(self, start_time=None):
    args = demisto.args()
    
    end_time = datetime.now()
    
    interval_time = end_time - timedelta(minutes=demisto.params().get('incidentFetchInterval', 360))
    
    if start_time is None or start_time < interval_time:
        start_time = interval_time
```